--- origin/CMakeLists.txt	2017-12-05 22:54:03.497116919 +0300
+++ src/CMakeLists.txt	2017-12-06 00:58:45.574908145 +0300
@@ -247,3 +247,69 @@
     target_link_libraries(minigzip64 zlib)
     set_target_properties(minigzip64 PROPERTIES COMPILE_FLAGS "-D_FILE_OFFSET_BITS=64")
 endif()
+
+#============================================================================
+# Conan patch
+# Dmitriy Vetutnev, ODNAT 2017
+#============================================================================
+
+# Patch PDB names
+if(WIN32 AND MSVC)
+    set_target_properties(zlib PROPERTIES PDB_NAME_DEBUG zlibd)
+    set_target_properties(zlibstatic PROPERTIES COMPILE_PDB_NAME_DEBUG zlibstaticd)
+endif()
+
+# Add minizip
+option(ENABLE_MINIZIP "Enable minizip" OFF)
+if(ENABLE_MINIZIP)
+    set(MINIZIP_DIR ${CMAKE_CURRENT_SOURCE_DIR}/contrib/minizip)
+    set(MINIZIP_SOURCES
+        ${MINIZIP_DIR}/ioapi.c
+        ${MINIZIP_DIR}/mztools.c
+        ${MINIZIP_DIR}/unzip.c
+        ${MINIZIP_DIR}/zip.c
+    )
+    set(MINIZIP_PUBLIC_HEADERS
+        ${MINIZIP_DIR}/crypt.h
+        ${MINIZIP_DIR}/ioapi.h
+        ${MINIZIP_DIR}/mztools.h
+        ${MINIZIP_DIR}/unzip.h
+        ${MINIZIP_DIR}/zip.h
+    )
+    if(WIN32)
+        list(APPEND MINIZIP_SOURCES ${MINIZIP_DIR}/iowin32.c)
+        list(APPEND MINIZIP_SOURCES ${MINIZIP_DIR}/iowin32.h)
+    endif()
+
+    add_library(minizip SHARED ${MINIZIP_SOURCES})
+    add_library(minizipstatic STATIC ${MINIZIP_SOURCES})
+
+    if(WIN32 AND MSVC)
+        set_target_properties(minizip PROPERTIES PDB_NAME_DEBUG minizipd)
+        set_target_properties(minizipstatic PROPERTIES COMPILE_PDB_NAME_DEBUG minizipstaticd)
+    endif()
+
+    set_target_properties(minizip PROPERTIES DEFINE_SYMBOL ZLIB_DLL)
+    set_target_properties(minizip PROPERTIES SOVERSION 1)
+
+    if(NOT CYGWIN)
+        set_target_properties(minizip PROPERTIES VERSION "1.0.0")
+    endif()
+
+    if(UNIX)
+        set_target_properties(minizip minizipstatic PROPERTIES OUTPUT_NAME minizip)
+    elseif(BUILD_SHARED_LIBS AND WIN32)
+        set_target_properties(minizip PROPERTIES SUFFIX "1.dll")
+    endif()
+
+    if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
+        install(TARGETS minizip minizipstatic
+            RUNTIME DESTINATION ${INSTALL_BIN_DIR}
+            ARCHIVE DESTINATION ${INSTALL_LIB_DIR}
+            LIBRARY DESTINATION ${INSTALL_LIB_DIR}
+        )
+    endif()
+    if(NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL )
+        install(FILES ${MINIZIP_PUBLIC_HEADERS} DESTINATION ${INSTALL_INC_DIR}/minizip)
+    endif()
+endif()
